service/src/main/java/org/ehrbase/service/EhrServiceImp.java

```diff
@@ -58,22 +57,35 @@
 @Service
 @Transactional()
 public class EhrServiceImp extends BaseService implements EhrService {
-    public static final String MODIFIABLE = "modifiable";
-    public static final String QUERYABLE = "queryable";
-    public static final String SUBJECT_ID = "subjectId";
-    public static final String SUBJECT_NAMESPACE = "subjectNamespace";
     public static final String DESCRIPTION = "description";
     private Logger logger = LoggerFactory.getLogger(getClass());
+    private final ValidationService validationService;
 
     @Autowired
-    public EhrServiceImp(KnowledgeCacheService knowledgeCacheService, DSLContext context, ServerConfig serverConfig) {
-
+    public EhrServiceImp(KnowledgeCacheService knowledgeCacheService, ValidationService validationService, DSLContext context, ServerConfig serverConfig) {
         super(knowledgeCacheService, context, serverConfig);
+        this.validationService = validationService;
     }
 
     @Override
     public UUID create(EhrStatus status, UUID ehrId) {
 
+        try {
+            validationService.check(status);
+        } catch (Exception e) {
+            // rethrow if this class, but wrap all others in InternalServerException
+            if (e.getClass().equals(UnprocessableEntityException.class))
+                throw (UnprocessableEntityException) e;
+            if (e.getClass().equals(IllegalArgumentException.class))
+                throw new ValidationException(e);
+            if (e.getClass().equals(ValidationException.class))
+                throw e;
+            else if (e.getClass().equals(org.ehrbase.validation.constraints.wrappers.ValidationException.class))
```

Comment:
We have two kinds of `ValidationException`, which is confusing. Do you know if we could replace the module specific `org.ehrbase.validation.constraints.wrappers.ValidationException` one with our project-wide `org.ehrbase.api.exception.ValidationException` one? I mean not just here, but generally?